src/main/java/de/tum/in/ase/parser/strategy/ParserContext.java

```diff
@@ -37,9 +42,11 @@ public Report getReport(File file) throws UnsupportedToolException, ParsingExcep
             return createFileTooLargeReport(file.getName());
         }
 
-        Builder parser = new Builder();
-        Document doc = parser.build(file);
-        return parseDocument(doc);
+        final DocumentBuilderFactory domFactory = DocumentBuilderFactory.newInstance();
```

Comment:
can you move the creation into a static method in XmlFileUtils (the same way as we do it in the Artemis code?

Reply:
and then call this method here, so that the builder is created each time from scratch (I think this is easier to handle for the garbage collector to free the memory afterwards)